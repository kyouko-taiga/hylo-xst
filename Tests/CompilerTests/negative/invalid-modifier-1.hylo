//! no-std

static trait P {

  public type X

}

static struct A {

  static typealias Z = A

}

public extension A {

  public static public fun f0() {}

  static public fun f0() {}

}

public fun main() {
  indirect var x: Void
}
