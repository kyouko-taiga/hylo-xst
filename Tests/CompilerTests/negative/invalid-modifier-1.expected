negative/invalid-modifier-1.hylo:3.1-7: error: declaration cannot be marked 'static'
static trait P {
~~~~~~
negative/invalid-modifier-1.hylo:5.3-9: error: declaration cannot be marked 'public'
  public type X
  ~~~~~~
negative/invalid-modifier-1.hylo:9.1-7: error: declaration cannot be marked 'static'
static struct A {
~~~~~~
negative/invalid-modifier-1.hylo:11.3-9: error: declaration cannot be marked 'static'
  static typealias Z = A
  ~~~~~~
negative/invalid-modifier-1.hylo:15.1-7: error: declaration cannot be marked 'public'
public extension A {
~~~~~~
negative/invalid-modifier-1.hylo:17.17-23: error: duplicate modifier
  public static public fun f0() {}
                ~~~~~~
negative/invalid-modifier-1.hylo:19.10-16: error: 'public' should occur before 'static'
  static public fun f0() {}
         ~~~~~~
negative/invalid-modifier-1.hylo:24.11: error: consecutive statements on the same line must be separated by ';'
  indirect var x: Void
          ^
